codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -294,13 +295,9 @@ private void generateOperationSerializerMiddleware(GenerationContext context, Op
             // delegate the setup and usage of the document serializer function for the protocol
             writeMiddlewareDocumentSerializerDelegator(model, symbolProvider, operation, generator, writer);
             writer.write("");
-
-            if (withRestBindings) {
-                writer.openBlock("if err := restEncoder.Encode(); err != nil {", "}", () -> {
-                    writer.write("return out, metadata, &aws.SerializationError{Err: err}");
-                });
-            }
-
+            writer.openBlock("if err := restEncoder.Encode(); err != nil {", "}", () -> {
+                writer.write("return out, metadata, &aws.SerializationError{Err: err}");
```

Comment:
If we're pulling everything else over, may as well pull over the serialization error too

Reply:
Will move this over.